pkg/clusterstack/factory.go

```diff
@@ -35,6 +34,11 @@ type Factory struct {
 	RunImageRef   string
 }
 
+type IPrinter interface {
```

Comment:
Putting "I" on an interface is not idiomatic Go. Generally Go just does verb + "er" or verb + "able" for interfaces.

Reply:
I was aware of the Java feel to it (heee) .. but was not sure how to avoid `Printer Printer` in the Factory struct. Any ideas?

Reply:
went with `Printer Printer`